packages/task/src/node/task.ts

```diff
@@ -17,19 +17,21 @@
 import { injectable } from 'inversify';
 import { ILogger, Emitter, Event, MaybePromise } from '@theia/core/lib/common/';
 import { TaskManager } from './task-manager';
-import { TaskInfo, TaskExitedEvent, TaskConfiguration } from '../common/task-protocol';
+import { TaskInfo, TaskExitedEvent, TaskConfiguration, TaskOutputEvent, TaskOutputProcessedEvent } from '../common/task-protocol';
 
 export interface TaskOptions {
-    label: string,
-    config: TaskConfiguration
-    context?: string
+    label: string;
+    config: TaskConfiguration;
+    context?: string;
 }
 
 @injectable()
```

Comment:
What is the lifecylce of a task? application-wide? if not then should not it be disposable and be disposed to release listeners and emitters?

Reply:
what do you think about this change ? https://github.com/theia-ide/theia/pull/5024/commits/fb1e1cbf5e565a8fb43b746499487f7a10427433

`Task` model class now implements Disposable, and will be disposed on `Task.onExit()`
@akosyakov @marechal-p